lib/random.php

```diff
@@ -198,67 +209,33 @@ function random_int($min, $max)
              * Provided by @CodesInChaos
              * 
              * @ref https://gist.github.com/CodesInChaos/03f9ea0b58e8b2b8d435
-             * 
-             * The rejection probability is at most 0.5, so this corresponds
-             * to a failure probability of 2^-128 for a working RNG
              */
-            for ($attempts = 0; $attempts < 128; $attempts++) {
-                /**
-                 * Generate a random integer...
-                 */
-                $rval = random_bytes(PHP_INT_SIZE);
-                if ($rval === false) {
-                    throw new Exception(
-                        'Random number generator failure'
-                    );
-                }
-                $value = 0;
-                for ($i = 0; $i < PHP_INT_SIZE; ++$i) {
-                    $value = ($value << 8) | ord($rval[$i]);
-                }
-                if ($value >= $min && $value <= $max) {
-                    /**
-                     * This value is within the accepted bounds.
-                     */
-                    return $value;
-                }
-            }
-            throw new Exception(
-                'random_int: RNG is broken - too many rejections'
-            );
-        }
-        /**
-         * We incremented $range earlier to test for overflows
-         */
-        --$range;
-
-        /**
-         * Initialize variables to 0
-         */
-        $attempts = $bits = $bytes = $mask = 0;
+            $bytes = PHP_INT_SIZE;
+            $mask = -1;
+        } else {
+            /**
+             * We incremented $range earlier to test for overflows
+             */
+            --$range;
 
-        $tmp = $range;
-        /**
-         * We want to store:
-         * $bytes => the number of random bytes we need
-         * $mask => an integer bitmask (for use with the &) operator
-         *          so we can minimize the number of discards
-         * 
-         * $bits is effectively ceil(log($range, 2)) without dealing with 
-         * type juggling
-         */
-        while ($tmp > 0) {
-            if ($bits % 8 === 0) {
-               ++$bytes;
+            /**
+             * $bits is effectively ceil(log($range, 2)) without dealing with 
+             * type juggling
+             */
+            while ($range > 0) {
+                if ($bits % 8 === 0) {
+                   ++$bytes;
+                }
+                ++$bits;
+                $range >>= 1;
```

Comment:
I'm undecided if there should be an additional check. This is a kind of heap paradox in my internal metaphysics of programming: How far away does a property need to be from the code that checks it to warrant an additional check?

Some will say, "this is easily implied by the checks above, so you're adding useless code" and others will say "but what if the checks above are broken in some way?"... I'm leaning slightly towards the latter.